src/App/Pages/Vault/GroupingsPage/GroupingsPageViewModel.cs

```diff
@@ -564,6 +642,42 @@ private async Task LoadDataAsync()
             }
         }
 
+        private bool IsVaultFilterMyVault => 
+            _vaultFilterSelection != null && _vaultFilterSelection == AppResources.MyVault;
+
+        private bool IsVaultFilterOrgVault =>
+            _vaultFilterSelection != null && _vaultFilterSelection != AppResources.AllVaults &&
+            _vaultFilterSelection != AppResources.MyVault;
+
+        private async Task<string> GetVaultFilterOrgIdAsync()
+        {
+            foreach (var org in _organizations)
+            {
+                if (org.Name == _vaultFilterSelection)
+                {
+                    return org.Id;
+                }
+            }
+            return null;
+        }
+
+        private List<FolderView> BuildFolders(List<FolderView> decFolders)
+        {
+            List<FolderView> folders = null;
+            foreach (var folder in decFolders)
+            {
+                if (_allCiphers.Any(c => c.FolderId == folder.Id))
+                {
+                    if (folders == null)
+                    {
+                        folders = new List<FolderView>();
+                    }
+                    folders.Add(folder);
+                }
+            }
+            return folders;
```

Comment:
This can be rewritten as:
```c#
var folders = decFolders.Where(f => _allCiphers.Any(c => c.FolderId == folder.Id)).ToList();
return folders.Any() ? folders : null;
```